skimage/segmentation/fast_draw.py

```diff
@@ -0,0 +1,336 @@
+        cg' and 'cg_mg' modes.
+    return_full_prob : bool, default False
+        If True, the probability that a pixel belongs to each of the labels
+        will be returned, instead of boolean array.
+    
+    See also
+    --------
+    skimage.segmentation.random_walker: random walker segmentation
+        The original random walker algorithm.
+        
+    References
+    ----------
+    [1] H.-E. Gueziri, L. Lakhdar, M. J. McGuffin and C. Laporte,
+    "FastDRaW - Fast Delineation by Random Walker: application to large
+    images", MICCAI Workshop on Interactive Medical Image Computing (IMIC),
+    Athens, Greece, (2016).
+    
+    Examples
+    --------
+    >>> from skimage.data import coins
+    >>> import matplotlib.pyplot as plt
+    >>> image = coins()
+    >>> labels = np.zeros_like(image)
+    >>> labels[[129, 199], [155, 155]] = 1 # label some pixels as foreground
+    >>> labels[[162, 224], [131, 184]] = 2 # label some pixels as background
+    >>> fastdraw = FastDRaW(image, beta=100, downsampled_size=100)
+    >>> segm = fastdraw.update(labels)
+    >>> plt.imshow(image,'gray')
+    >>> plt.imshow(segm, alpha=0.7)
+    """
+    
+    def __init__(self, image, beta=300, downsampled_size=100,
+                 tol=1.e-3, return_full_prob=False):
+        
+        assert (beta > 0), 'beta should be positive.'
+        self.beta = beta
+        self.return_full_prob = return_full_prob
+        self.image = rgb2grey(image)
+        ## It is important to normalize the data between [0,1] so that beta
+        ## makes sens
+        self.image = img_as_float(self.image)
+        
+        if downsampled_size > min(self.image.shape):
+            warn('The size of the downsampled image if larger than the '
+                 'original image. The computation time could be affected.')
+        
+        ## Compute the graph's Laplacian for both the full resolution `L` 
+        ## and the down-sampled `ds_L` images
+        self.L = self._buildGraph(self.image)
+        ratio = float(self.image.shape[0])/self.image.shape[1]
+        self.dim = (int(downsampled_size*ratio), downsampled_size)
+        self.ds_image = resize(self.image, self.dim, order=0,
+                                 preserve_range=True)
+        self.ds_L = self._buildGraph(self.ds_image)
+        
+        ## Initialize the ROI to zero
+        self.ds_maskROI = np.zeros_like(self.ds_image, dtype=np.bool)
+        ## `full_to_ds_ratio` is used to convert labels from full resolution
+        ## image to down-sampled image
+        self.full_to_ds_ratio = (float(self.dim[0])/self.image.shape[0],
+                                 float(self.dim[1])/self.image.shape[1])
+
+    def _buildGraph(self, image):
+        """Builds the graph: vertices, edges and weights from the `image` 
+        and returns the graph's Laplacian `L`.
+        """
+        # Building the graph: vertices, edges and weights
+        nsize = reduce(lambda x, y : x*y, image.shape,1)
```

Comment:
This is the cause of the test problem, `reduce` is not imported or defined. I'm surprised it gets to the doctests - this should fail in the usual test suite.